### PR TITLE
LPS-135069 Add subscription to tags in Questions

### PR DESCRIPTION
**Pull request by LuismiBarcos**
We are adding a new feature in the Questions App: the subscription to the tags of the application.
To do that, we added a new subscription button to the tags as as shown in the following picture
image

Right now, the background of the button doesn't change when it is clicked because the component is not re-rendered. As you can see in the file [Tag.es.js](https://github.com/liferay-frontend/liferay-portal/compare/master...LuismiBarcos:LPS-135069?expand=1#diff-eded3d52677ef912dc8ab8f2389fce5a3cd273fa30aaccc3ea6a31ff22c5c2b6) to re-render the icon it would be necessary to re-render all the list of tags.

IMHO this is not a good practice. Maybe it would not have a big impact right now but there will be a lot of not-needed renders and I don't know if this will be a problem in the future.

I think that the best option is to create a "dumb" component to do the subscription. I say is a "dumb" component because it will need the actions to do when it is clicked. The component only will have the responsibility of changing the background to provided feedback.

Also, there are two more components to manage the subscriptions in other features on Questions. These components are [Subsription.es.js](https://github.com/liferay/liferay-portal/blob/master/modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/Subscription.es.js) and [SectionSubscription.es.js](https://github.com/liferay/liferay-portal/blob/master/modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/SectionSubscription.es.js) with similar code that I think we can unify. This would help with that.

Of course, do this will require more effort, and maybe I'm getting things too complicated or there is a better solution that I don't see. Because of that, I'm here to discuss what could be the better solution. 🙂

### CODE REVIEW

**Comment by LuismiBarcos** [comment]
Added some changes with my proposal. I have to review the code (mostly the naming) but in general, this is my idea, use just one component for the subscriptions.

Missing fix the redirect from notifications of the tags to the proper question. I am on it.

**Comment by LuismiBarcos** [comment]
ci:test:relevant

**Comment by LuismiBarcos** [comment]
ci:test:relevant

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/SubscriptionButton.es.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayIcon from '@clayui/icon';
+import {useMutation} from 'graphql-hooks';
+import React, {useEffect, useState} from 'react';
+
+export default ({
+	isSubscribed,
```
Comment (nhpatt) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/SubscriptionButton.es.js:21]: This has a very huge amount of props, not usually a good sign

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/SubscriptionButton.es.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	onSubscription,
+	parentSection = false,
+	queryVariables,
+	showTitle = false,
+	subscribeQuery,
```
Comment (nhpatt) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/components/SubscriptionButton.es.js:26]: Why pass the queries?

File: modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js [hunk 3]
```diff
@@ -587,7 +590,15 @@ export default withRouter(
 												}
 											);
 										}}
-										section={section}
+										parentSection={section.parentSection}
+										queryVariables={{
+											messageBoardSectionId: section.id,
+										}}
+										showTitle={true}
+										subscribeQuery={subscribeSectionQuery}
```
Comment (nhpatt) [inline modules/apps/questions/questions-web/src/main/resources/META-INF/resources/js/pages/questions/Questions.es.js:598]: I see... passing the queries without a real contract in the subscription APIs seems risky. Sending a function or something like that would be nicer.

**Comment by nhpatt** [comment]
Let's refactor later if we consider it necessary.

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Pull request closed without merging**